### PR TITLE
Improve mfhdf tests

### PR DESCRIPTION
**Pull request by bmribler**
Moved long/cluttered test functions out of the main test file hdftest.c.

### CODE REVIEW

File: mfhdf/test/tattributes.c [hunk 1]
```diff
@@ -79,7 +80,7 @@
 #define ATTR_LEN_ZERO 0
 
 static intn
-test_count(void)
+test_count()
```
Comment (derobins) [inline mfhdf/test/tattributes.c:83]: C should have the void keyword

**Comment by bmribler** [comment]
It was the failure like this one, which I've seen it last year and I asked @byrnHDF about it, but we couldn't figure out what it was:

Test  #66: MFHDF_TEST-hdftest ...........................***Failed  Required regular expression not found. Regex=[HDF-SD test passes
]
 
I googled, some people said something about the CMake version.  They reinstalled CMake
and the problem was gone.  Have you heard anything about that?  But all the other t
ests passed!

**Comment by derobins** [comment]
It's weird because it passes everywhere else, the line `set (passRegex "HDF-SD test passes")` is clearly set in CMakeTests.cmake, and you didn't change anything regarding CMake.

**Review by derobins — APPROVED** [review]

**Comment by derobins** [comment]
The problem was a missing return statement on a function that returned an int, so apparently on MacOS it returned non-zero trash, which was interpreted as errors.

**Comment by bmribler** [comment]
Thank you!

**Pull request merged**